### PR TITLE
Increase coverage in iot_mqtt_api.c

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_platform.c [hunk 1]
```diff
@@ -0,0 +1,306 @@
+/*
+ * IoT MQTT V2.1.0
+ * Copyright (C) 2018 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_platform.c]: Wondering what the rule we follow is for this date, as some files in the repo are 2018 and others are 2019. Should this be 2020?

Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_platform.c]: Updated.

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 2]
```diff
@@ -1139,29 +1184,37 @@ TEST( MQTT_Unit_API, DisconnectMallocFail )
...
     /* 'disconnected' flag should be set */
-    TEST_ASSERT_EQUAL( true, mqttConnection->disconnected );
+    TEST_ASSERT_EQUAL( true, _pMqttConnection->disconnected );
 
-    /* Make sure reference count is decremented when 'disconnected'
-     * connection is passed without any attempts to close socket again. */
-    IotMqtt_Disconnect( mqttConnection, 0 );
-    TEST_ASSERT_EQUAL_INT( 1, mqttConnection->references );
+    /* Attempt to use a closed connection. */
+    publishInfo.pTopicName = "test/";
```
Comment (dcgaws) [inline libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: Similar to above, in re fewer hard coded literals.

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 3]
```diff
@@ -816,6 +816,51 @@ TEST( MQTT_Unit_API, OperationCreateDestroy )
...
+
+    /* Allocate an operation for an incoming publish. */
+    pOperation = IotMqtt_MallocOperation( sizeof( _mqttOperation_t ) );
+    TEST_ASSERT_NOT_NULL( pOperation );
+    ( void ) memset( pOperation, 0x00, sizeof( _mqttOperation_t ) );
+
+    pOperation->incomingPublish = true;
+    pOperation->pMqttConnection = _pMqttConnection;
+    pOperation->u.publish.publishInfo.pTopicName = "test/";
+    pOperation->u.publish.publishInfo.topicNameLength = 5;
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: Fixed

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 4]
```diff
@@ -816,6 +816,51 @@ TEST( MQTT_Unit_API, OperationCreateDestroy )
...
+    pOperation->u.publish.pReceivedData = IotMqtt_MallocMessage( 1 );
+    pOperation->u.publish.publishInfo.pPayload = pOperation->u.publish.pReceivedData;
+    pOperation->u.publish.publishInfo.payloadLength = 1;
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: fixed

File: libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c [hunk 5]
```diff
@@ -816,6 +816,51 @@ TEST( MQTT_Unit_API, OperationCreateDestroy )
...
+
+    /* Increment the MQTT connection's reference count to prevent it from being destroyed
+     * until the test is over. */
+    _pMqttConnection->references += 2;
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c]: Each call to IotMqtt_Disconnect decrements this by 1.  There are 2 calls to IotMqtt_Disconnect following this line.

File: libraries/standard/mqtt/test/system/iot_tests_mqtt_system.c [hunk 6]
```diff
@@ -580,6 +580,8 @@ static void _subscribePublishWait( IotMqttQos_t qos )
             publishInfo.topicNameLength = ( uint16_t ) strlen( publishInfo.pTopicName );
             publishInfo.pPayload = _pSamplePayload;
             publishInfo.payloadLength = _samplePayloadLength;
+            publishInfo.retryLimit = 3;
```
[the remark this answers is not in the document]
Comment (gordonwang0) [reply libraries/standard/mqtt/test/system/iot_tests_mqtt_system.c]: added config

**Pull request merged**